libcamera_app.hpp

```diff
@@ -719,6 +722,25 @@ class LibcameraApp
 			preview_->Show(fd, size, w, h, stride);
 		}
 	}
+	void configureDenoise(bool video_mode)
+	{
+		if (options.denoise == "auto") {
+			controls_.set(controls::draft::NoiseReductionMode,
+				      video_mode ? libcamera::controls::draft::NoiseReductionModeFast
+						 : libcamera::controls::draft::NoiseReductionModeHighQuality);
+		} else {
+			static std::map<std::string, int> denoise_table = {
```

Comment:
This can be `const` and I would probably also move it to the beginning of the function. Since it's static it shouldn't make a difference, no?

Reply:
I have reworked this function with this and your other suggestions.